rollup.config.js

```diff
@@ -5,56 +5,65 @@ import commonjs from 'rollup-plugin-commonjs'
 import babel from 'rollup-plugin-babel'
 import json from 'rollup-plugin-json'
 import flow from 'rollup-plugin-flow'
-import uglify from 'rollup-plugin-uglify'
+import { terser } from 'rollup-plugin-terser'
 import visualizer from 'rollup-plugin-visualizer'
 import sourceMaps from 'rollup-plugin-sourcemaps'
 import ignore from 'rollup-plugin-ignore'
 import pkg from './package.json'
 
 const cjs = {
-  format: 'cjs',
   exports: 'named',
+  format: 'cjs',
+  sourcemap: true,
+}
+
+const esm = {
+  format: 'esm',
+  sourcemap: true,
 }
 
 const commonPlugins = [
   flow({
     // needed for sourcemaps to be properly generated
     pretty: true,
   }),
+  sourceMaps(),
   json(),
   nodeResolve(),
-  sourceMaps(),
+  babel({
+    exclude: 'node_modules/**',
+    plugins: ['external-helpers'],
+  }),
   commonjs({
     ignoreGlobal: true,
     namedExports: {
       'react-is': ['isValidElementType'],
     },
   }),
-  babel({
-    plugins: ['external-helpers'],
-  }),
   replace({
     __DEV__: JSON.stringify(false), // disable flag indicating a Jest run
   }),
 ]
 
+const globals = { 'prop-types': 'PropTypes', react: 'React' }
+
 const configBase = {
   input: 'src/index.js',
-  globals: { 'prop-types': 'PropTypes', react: 'React' },
   external: ['react', 'prop-types'].concat(
     Object.keys(pkg.dependencies),
     Object.keys(pkg.peerDependencies)
   ),
   plugins: commonPlugins,
-  sourcemap: true,
 }
 
 const umdBaseConfig = Object.assign({}, configBase, {
   output: {
+    exports: 'named',
     file: 'dist/styled-components.js',
     format: 'umd',
+    globals,
     name: 'styled',
-    exports: 'named',
+    sourcemap: true,
   },
   external: ['react'],
```

Comment:
To mirror globals (they are mismatched now) it's better to use `Object.keys(globals)`

Reply:
Hmm this would be a breaking change maybe... since `PropTypes` was getting packaged before.

Reply:
Then just remove `prop-types` from globals.

Reply:
Wait why are we packaging PropTypes? Let's not do that?

Reply:
Not sure, but I added it to the 4.0 roadmap since it would be a breaking change.

Reply:
Ace, thanks!